### PR TITLE
Overload function 'getDocument' to accept identifier as Int too

### LINKED ISSUE #38: When asking for the document identifier, only string is accepted but Int should be accepted too
## Problem 
When using a route like `getDocument`, one of the parameter is `identifier`. This parameter is of type `string`. Problem is, often the `id` in documents are of type `Int`. 

Because of that the user has to change types before using that route. 

### Example
In the following example I have to change my type from Int to String to my my request work.
```swift
private struct Movie: Codable, Equatable {
    let id: Int
    let title: String
    let comment: String?

    enum CodingKeys: String, CodingKey {
        case id
        case title
        case comment
    }
    init(id: Int, title: String, comment: String? = nil) {
        self.id = id
        self.title = title
        self.comment = comment
    }

}

let movie = Movie(id: 1, title: "test", comment: "no comment")
client.getDocument(UID: "movies", identifier: String(movie.id))
```

## Solution

We should accept both types.

```swift
public func getDocument<T>(
        UID: String,
        identifier: String | Int,
        _ completion: @escaping (Result<T, Swift.Error>) -> Void)
        where T: Codable, T: Equatable {
        self.documents.get(UID, identifier, completion)
    }
```

### PR DESCRIPTION
**Pull request by ednofedulo**
Fixes #38

### CODE REVIEW

**Comment by ednofedulo** [comment]
missing test

File: Sources/MeiliSearch/Client.swift
Comment (ppamorim) [inline Sources/MeiliSearch/Client.swift:214]: @ednofedulo Is this correct to transform the `Int` into a `String`?

Comment (ppamorim) [reply Sources/MeiliSearch/Client.swift:214]: I don't see any aditional risk by doing this change, specially by the fact that this value is used by this:

```swift
let query: String = "/indexes/\(UID)/documents/\(identifier)"
```

Comment (ednofedulo) [reply Sources/MeiliSearch/Client.swift:214]: I don't see any risk involving this parsing, since the values of a Int variable is very limited compared to a String, which the function already accepts without any validation of value.

**Review by ppamorim — APPROVED** [review]

**Comment by ppamorim** [comment]
@curquiza Could you please review and merge if approved?

File: Sources/MeiliSearch/Client.swift
Comment (bidoubiwa) [reply Sources/MeiliSearch/Client.swift:214]: Would this still work with an identifier that is a string? for example
```swift
 self.client.getDocument(
            UID: uid,
            identifier: "ab12cd"
        )
```

File: Tests/MeiliSearchIntegrationTests/DocumentsTests.swift
Comment (bidoubiwa) [inline Tests/MeiliSearchIntegrationTests/DocumentsTests.swift:145]: This tests is useless since it does not check if the error is due to a 404 or due to an internal error in the code. Could you change it with the following ?

```suggestion
    func testGetOneDocumentWithIntIdentifierAndSucceed() {
        let movie: Movie = Movie(id: 10, title: "test", comment: "test movie")
        let getExpectation = XCTestExpectation(description: "Get one document using and int identifier and succeed")
        self.client.getDocument(
                   UID: uid,
                   identifier: 10
               ) { (result: Result<Movie, Swift.Error>) in

                   switch result {
                   case .success(let returnedMovie):
                       XCTAssertEqual(movie, returnedMovie)
                   case .failure(let error):
                       XCTFail(error.localizedDescription)
                   }
                   expectation.fulfill()

               }
    }
```

File: Tests/MeiliSearchIntegrationTests/DocumentsTests.swift
Comment (bidoubiwa) [inline Tests/MeiliSearchIntegrationTests/DocumentsTests.swift:401]:
```suggestion
        ("testGetOneDocumentWithIntIdentifierAndSucceed", testGetOneDocumentWithIntIdentifierAndSucceed),
```

**Review by bidoubiwa — CHANGES_REQUESTED** [review]
I changed the tests to make it relevant

**Comment by curquiza** [comment]
Hello @ednofedulo! Thanks for your PR.
I see you do this PR from your master branch which we do not recommend 🙂 For example, if we have refused your PR, you would have a broken `master` branch compared to ours. Let's keep your PR as it is now, it's okay, but next time, I recommend you creating a separate branch.
See here our [contribution guidelines](https://github.com/meilisearch/meilisearch-swift/blob/master/CONTRIBUTING.md#how-to-contribute).

I'm reviewing your PR soon!

File: Sources/MeiliSearch/Client.swift
Comment (ednofedulo) [reply Sources/MeiliSearch/Client.swift:214]: the actual behaviour that accepts string is still as is

**Comment by ednofedulo** [comment]
Hi, @curquiza !
Sorry, i missed that step. Actually im still learning how this PR thing works here on Github. Thanks for your advice.

**Comment by curquiza** [comment]
@ednofedulo Not a problem! We are all learners here! 🙂
I don’t know if you are already on [our Slack](https://slack.meilisearch.com/), feel free to joins us, it would be easier to discuss about contribution (I’m Clementine). And we also have a #contributions channel if you are interested in MeiliSearch’s tools contributions in general!

**Comment by curquiza** [comment]
bors try

**Comment by curquiza** [comment]
Hello @ednofedulo, the tests `testUpdateAndGetDocuments` is failing in `DocumentsTests`

```bash
/Users/runner/work/meilisearch-swift/meilisearch-swift/Tests/MeiliSearchIntegrationTests/DocumentsTests.swift:143: error: -[MeiliSearchIntegrationTests.DocumentsTests testUpdateAndGetDocuments] : failed - The operation couldn’t be completed. (MeiliSearch.MSError error 1.)
```

**Comment by ednofedulo** [comment]
> Hello @ednofedulo, the tests `testUpdateAndGetDocuments` is failing in `DocumentsTests`
> 
> ```shell
> /Users/runner/work/meilisearch-swift/meilisearch-swift/Tests/MeiliSearchIntegrationTests/DocumentsTests.swift:143: error: -[MeiliSearchIntegrationTests.DocumentsTests testUpdateAndGetDocuments] : failed - The operation couldn’t be completed. (MeiliSearch.MSError error 1.)
> ```

Hi @curquiza ! Fixed it.

**Review by curquiza — APPROVED** [review]
Awesome! Thanks!

**Comment by curquiza** [comment]
bors try

**Comment by curquiza** [comment]
bors merge

**Review by ppamorim — APPROVED** [review]

**Comment by curquiza** [comment]
@bidoubiwa, we need your approval (if you are ok with the change) to apply the `bors merge`.

**Comment by ppamorim** [comment]
bors merge

**Comment by ppamorim** [comment]
@bidoubiwa Sorry I forgot about your review and bors blocked the merge. We are waiting for you.

**Review by bidoubiwa — APPROVED** [review]

**Comment by ppamorim** [comment]
bors merge

**Comment by ppamorim** [comment]
Thank you @ednofedulo!

**Pull request merged**
